system/lib/wasmfs/file.h

```diff
@@ -97,6 +98,10 @@ class File : public std::enable_shared_from_this<File> {
   File(FileKind kind, mode_t mode, backend_t backend)
     : kind(kind), mode(mode), backend(backend) {
     atime = mtime = ctime = time(NULL);
+
+    struct timespec ts;
+    clock_gettime(CLOCK_REALTIME, &ts);
+    atime_nsec = mtime_nsec = ctime_nsec = ts.tv_nsec;
```

Comment:
It is more comprehensive to support nanoseconds as well, and I think this the right way to do it (but we could also get the non-nanosecond times from this as well, to save the previous `time(..)` call).

However, I think we need to consider if we want to add ns. Technically a POSIX-compliant FS doesn't need that (I don't think POSIX guarantees a particular precision). And on the Web the timing info we get from `Date.now()` etc. may have limited resolution anyhow.

And, adding ns support has a cost in terms of code size. So overall I lean towards not adding it, and just documenting that decision.

@tlively what do you think?

Reply:
+1 to removing the previous `time` call.

I would lean toward keeping this. I don't think it's too much of a stretch to imagine an Emscripten application that would want at least millisecond precision in the time stamps. For example, anything doing something similar to `make` to figure out when file changes mean something needs to be recomputed would need more than second precision.

I also expect that the code size increase would be fairly small.